src/tiled/createobjecttool.cpp

```diff
@@ -195,7 +195,7 @@ void CreateObjectTool::finishNewMapObject()
     Q_ASSERT(mNewMapObjectItem);
 
     ObjectGroup *objectGroup = currentObjectGroup();
-    if (!objectGroup) {
+    if (!objectGroup || !objectGroup->isUnlocked()) {
```

Comment:
The check is too late at this point. It should not be possible to start the creation of a new object.